lib/foreman/renderer.rb

```diff
@@ -57,6 +59,11 @@ def foreman_url(action = "built")
               :token     => (@host.token.value unless @host.try(:token).nil?), :kind => action
     end
 
+    def foreman_server_fqdn
+      config = URI.parse(Setting[:unattended_url])
+      config.host || request.host
```

Comment:
Yeah, FQDN is not enough, but it's a good start. Scheme is always https in our case and port is either 443 or 8443 depending on connection type. I filed a ticket that could provide the rest via settings: http://projects.theforeman.org/issues/14245. And if we can get full URL with scheme and port, at least for Discovery case we can detect what we need.